File: Structural/Bridge/Cpp/BasicRemote.cpp
// BasicRemote.cpp

#include "BasicRemote.h"
#include <iostream>

BasicRemote::BasicRemote(IDevice* device) : IRemote(device) {}

void BasicRemote::togglePower() {
  if (device->isOn()) {
    device->turnOff();
  } else {
    device->turnOn();
  }
}

void BasicRemote::volumeUp() {
  device->setVolume(device->getVolume() + 10);
}

void BasicRemote::volumeDown() {
  device->setVolume(device->getVolume() - 10);
}

File: Structural/Bridge/CSharp/BasicRemote.cs
// BasicRemote.cs

public class BasicRemote : IRemote {
  protected IDevice device;

  public BasicRemote(IDevice device) {
    this.device = device;
  }

  public void TogglePower() {
    if (device.IsOn()) {
      device.TurnOff();
    } else {
      device.TurnOn();
    }
  }

  public void VolumeUp() {
    device.SetVolume(device.GetVolume() + 10);
  }

  public void VolumeDown() {
    device.SetVolume(device.GetVolume() - 10);
  }
}
